bin/util/common.sh

```diff
@@ -87,10 +121,25 @@ export_env_dir() {
 curl_retry_on_18() {
 	local ec=18;
 	local attempts=0;
-	while [[ $ec -eq 18 && $attempts -lt 3 ]]; do
-		((attempts++))
+	while (( ec == 18 && attempts++ < 3 )); do
 		curl "$@" # -C - would return code 33 if unsupported by server
 		ec=$?
 	done
 	return $ec
 }
+
+err_trap() {
+	error <<-EOF
+		An unknown internal error occurred.
+	
+		Contact Heroku Support for assistance if this problem persists.
+		
+		Stack trace follows for debugging purposes:
+		$(
+			local frame=0
+			while caller $frame; do
+				((frame++));
+			done
+		)
+	EOF
+}
```

Comment:
It's a "last resort" error trap, so if it triggers, I want people to open a ticket. The regular "external things like S3 may be down" cases are covered by explicit error handling cases. If the trap is hit, it typically won't be because of an external dependency.

Reply:
Just talking from a support perspective. We're highly highly discouraged from ever suggesting people open a ticket as their next action. It couldn't hurt to put a call out to status.heroku.com

Do you know how often this gets hit?